### PR TITLE
Add subscription as argument on update and error callbacks

### PR DESCRIPTION
**Pull request by lukeapage**

### CODE REVIEW

File: src/openapi/streaming/subscription.ts [hunk 1]
```diff
@@ -1159,7 +1163,7 @@ class Subscription {
         }
 
         // Serialization of Snapshot is not yet supported.
-        this.onUpdate?.(response.Snapshot, this.UPDATE_TYPE_SNAPSHOT);
+        this.onUpdate?.(response.Snapshot, this.UPDATE_TYPE_SNAPSHOT, this);
```
Comment (ikarami) [inline src/openapi/streaming/subscription.ts:1166]: nitpick - why those types are repeated within the class itself even though they are already defined outside as:

const updateTypes = {
    UPDATE_TYPE_SNAPSHOT: 1,
    UPDATE_TYPE_DELTA: 2,
} as const;

File: src/openapi/streaming/subscription.ts [hunk 2]
```diff
@@ -45,24 +45,28 @@ export interface StreamingOptions {
      * @param data - data received
      * @param updateType - either be subscription.UPDATE_TYPE_DELTA or subscription.UPDATE_TYPE_SNAPSHOT
      */
-    onUpdate?: (data: unknown, updateType: SubscriptionUpdateTypes) => void;
+    onUpdate?: (
+        data: unknown,
+        updateType: SubscriptionUpdateTypes,
+        subscription: Subscription,
```
Comment (ikarami) [inline src/openapi/streaming/subscription.ts:51]: please update JSDoc for every function that now accepts subscription as a parameter

**Review by ikarami — APPROVED** [review]
Two minor issues, but LGTM

File: src/openapi/streaming/subscription.ts (hunk 1 above)
Comment (lukeapage) [reply src/openapi/streaming/subscription.ts:1166]: I don't know but I don't have much of a preference myself. I guess this way it is consistent with how it is checked (consumers use subscription.UPDATE_TYPE_SNAPSHOT). I agree its a bit messy and would be better exposed separately, but  I think its a bigger standalone task.

File: src/openapi/streaming/subscription.ts (hunk 2 above)
Comment (lukeapage) [reply src/openapi/streaming/subscription.ts:52]: Fixed

**Pull request merged**